packages/gitbook/src/components/SitePage/SitePage.tsx

```diff
@@ -237,12 +257,76 @@ async function getPageDataWithFallback(args: {
 }) {
     const { context: baseContext, pagePathParams } = args;
     const { context, pageTarget } = await fetchPageData(baseContext, pagePathParams);
+    const pageMetaLinks = await (pageTarget?.page
```

Comment:
IMPORTANT: to avoid getting too many requests at once, do a % rolling here based on the site ID